xdsl/dialects/x86/assembly.py

```diff
@@ -106,3 +106,14 @@ def parse_type_pair(parser: Parser) -> SSAValue:
     parser.parse_punctuation(":")
     type = parser.parse_type()
     return parser.resolve_operand(unresolved, type)
+
+
+def masked_source_str(reg_in: SSAValue, mask: SSAValue, z: UnitAttr | None) -> str:
+    """
+    Tests printing format for the register followed by the {k} (and optionally {z})
```

Comment:
This docstring doesn't seem right? This function is used outside of testing right?